scripts/config.docker.yaml

```diff
@@ -139,21 +119,30 @@ mail:
 #     5. Click "Server key"
 #     6. Under "APIs & auth" click "YouTube Data API" and then click "Enable API"
 youtube-v3-key: '${YOUTUBE_KEY}'
-# Minutes between saving channel state to disk
-channel-save-interval: 5
 # Limit for the number of channels a user can register
 max-channels-per-user: 5
 # Limit for the number of accounts an IP address can register
 max-accounts-per-ip: 5
 # Minimum number of seconds between guest logins from the same IP
 guest-login-delay: 60
 
-# Configure statistics tracking
-stats:
-  # Interval (in milliseconds) between data points - default 1h
-  interval: 3600000
-  # Maximum age of a datapoint (ms) before it is deleted - default 24h
-  max-age: 86400000
+# Allows you to customize the path divider. The /r/ in http://localhost/r/yourchannel
+# Acceptable characters are a-z A-Z 0-9 _ and -
+channel-path: 'r'
+# Allows you to blacklist certain channels.  Users will be automatically kicked
+# upon trying to join one.
+channel-blacklist: []
+# Minutes between saving channel state to disk
+channel-save-interval: 5
+# Determines channel data storage mechanism.
+# Defaults to 'file', in which channel data is JSON stringified and saved to a file
+# in the `chandump/` folder.  This is the legacy behavior of CyTube.
+# The other possible option is 'database', in which case each key-value pair of
+# channel data is stored as a row in the `channel_data` database table.
+# To migrate legacy chandump files to the database, shut down CyTube (to prevent
+# concurrent updates), then run `node lib/channel-storage/migrate.js`.
+channel-storage:
+  type: '${CHANNEL_STORAGE}'
```

Comment:
Let's just force this to 'database'.  The flatfile format is flawed, outdated, and only remains because I haven't gotten around to writing the upgrade script to migrate existing servers yet (other than cytu.be's database).

Reply:
The only reason I chose to leave this in as an environment variable was with the intention to eventually implement SQLite, which would be more lightweight for a development instance. This is another instance, though, where I don't think having it available as a switch is a bad thing. That being said, I should change it to database in the Dockerfile.

Reply:
There is actually a separate -- stalled -- project to make the entire persistence layer pluggable for different DB engines such as SQLite and Postgres.  I'm a fan of SQLite and would love to support it, but at some point I set that refactoring project aside and never picked it up again.

Anyways, I think that is independent of `channel-storage`, which refers only to how the channel module data is stored (playlists, emotes, etc.) -- even when set to `file`, there are things stored in the DB (ranks, bans, ownership information, library).

Way back in the day, all the module data was in memory, and I lost it whenever the server crashed, so I put in a hotfix to dump it as JSON to disk.  Well, that hotfix lasted for years and didn't age well, leading to all kinds of issues with corrupted files, channels losing data when the OpenVZ container's host ran out of inodes, etc. and so I did a low-effort migration of this data from files to the database but sadly the work to finish the job, auto-migrate everyone else, and deprecate flatfile support fell by the wayside.